packages/api/src/external/commonwell/document/document-query.ts

```diff
@@ -553,21 +561,23 @@ export async function downloadDocsAndUpsertFHIR({
             },
           };
 
-          if (file.isNew && !ignoreFhirConversionAndUpsert) {
-            try {
-              await convertCDAToFHIR({
-                patient,
-                document: doc,
-                s3FileName: file.key,
-                s3BucketName: file.bucket,
-                requestId,
-              });
-            } catch (err) {
-              // don't fail/throw or send to Sentry here, we already did that on the convertCDAToFHIR function
-              log(
-                `Error triggering conversion of doc ${doc.id}, just increasing errorCountConvertible - ${err}`
-              );
-              errorCountConvertible++;
+          if (shouldConvertCDA) {
+            if (isFileConvertible) {
```

Comment:
The reason i didnt add `isFileConvertible` to the shouldConvertCDA is to maintain the same logic where if a file is not new we we increase the error count

Reply:
Sorry, but I don't understand. And looking at the lines 605-611 its even more confusing.

If I understand correctly, we only need to keep track of whether the current iteration/loop is convertible or not. How we determine that is a different thing: we could just say it is convertible if the downloaded file is convertible, without considering the doc ref.

Before we used the doc ref because the doc ref was the file being downloaded, but that's not the case anymore. So I think we can just update the logic to base the `isConvertible` on the file, not the doc ref.

Would that work?